File: GeneratedEntityReaderClasses/Components/VariableStorageComponent.cs
namespace GeneratedNoitaClasses;
public class VariableStorageComponent : NoitaComponentBase,  iNoitaType<VariableStorageComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString name { get; set; } = new();
	public NoitaString value_string { get; set; } = new();
	public NoitaInt value_int { get; set; } = new();
	public NoitaBool value_bool { get; set; } = new();
	public NoitaFloat value_float { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		name.Read(s);
		value_string.Read(s);
		value_int.Read(s);
		value_bool.Read(s);
		value_float.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		name.Write(s);
		value_string.Write(s);
		value_int.Write(s);
		value_bool.Write(s);
		value_float.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/VariableStorageComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class VariableStorageComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    name:NoitaString = field(default_factory=NoitaString)
    value_string:NoitaString = field(default_factory=NoitaString)
    value_int:NoitaInt = field(default_factory=NoitaInt)
    value_bool:NoitaBool = field(default_factory=NoitaBool)
    value_float:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.name.Read(s)
        self.value_string.Read(s)
        self.value_int.Read(s)
        self.value_bool.Read(s)
        self.value_float.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.name.Write(s)
        self.value_string.Write(s)
        self.value_int.Write(s)
        self.value_bool.Write(s)
        self.value_float.Write(s)

